nixos/tests/caddy.nix

```diff
@@ -150,6 +163,17 @@ import ./make-test-python.nix (
             )
             webserver.wait_for_open_port(80)
             webserver.succeed("curl http://localhost | grep caddy")
+
+        with subtest("invalid config shows warning when validation is not enforced"):
+            webserver.succeed(
+                "${invalidConfigWarning}/bin/switch-to-configuration test >&2"
+            )
+            webserver.wait_for_unit("caddy")
+
+        with subtest("invalid config fails build when validation is enforced"):
+            webserver.fail(
```

Comment:
todo: the build is expect to fail ... so i need to check against that differently ...